### PR TITLE
[fcos] Update UPI to Terraform 0.12.x

### PR DESCRIPTION
**Pull request by LorbusChris**
Sync with upi changes from master

Add a new tf template file for NetworkManager keyfiles 

/cc @vrutkovs

### CODE REVIEW

File: upi/vsphere/machine/ignition.tf [hunk 1]
```diff
@@ -24,12 +24,12 @@ data "ignition_file" "static_ip" {
...
   path       = "/etc/NetworkManager/system-connections/eth0"
-  mode       = "0600"
+  mode       = "384"
 
   content {
     content = <<EOF
 [connection]
-id=Wired connnection 1
+id=Wired connection 1
 uuid=27afa607-ee36-43f0-b8c3-9d245cdc4bb3
```
Comment (LorbusChris) [inline upi/vsphere/machine/ignition.tf]: I wonder whether there is a common way to autogenerate this with terraform? Hardcoding the uuid seems weird.

Comment (vrutkovs) [reply upi/vsphere/machine/ignition.tf]: Right, I think https://www.terraform.io/docs/configuration/functions/uuid.html can be used here

**Comment by LorbusChris** [comment]
/hold
I'll add the random uuid generation

**Comment by abhinavdahiya** [comment]
wouldn't it be better to fix this in master? how is this fcos specific.

**Comment by LorbusChris** [comment]
@abhinavdahiya this PR grew from a small fcos specific fixup (first commit). Porting over to master for full fledged testing right now!

**Comment by LorbusChris** [comment]
/hold
until https://github.com/openshift/installer/pull/3235 is in

**Pull request closed without merging**